BigBaseV2/src/script_mgr.hpp

```diff
@@ -12,8 +25,15 @@ namespace big
 
 		void add_script(std::unique_ptr<script> script);
 		void remove_all_scripts();
-
 		void tick();
+		std::vector<script*> get_scripts_data() {
+			std::vector<script*> retvalue{};
+			for (auto& s : m_scripts) { 
+				if(s->get_enabled_pointer())
+					retvalue.push_back(s.get());
+			}
+			return retvalue;
+		}
```

Comment:
Instead of creating a new vector you can return the entire script vector by reference, though then you'll have to filter through the scripts in the UI.

```cpp
std::vector<std::unique_ptr<script>>& get_scripts_data()
{
    return m_scripts;
}
```

Reply:
The reason for this little detour is to disregard scripts that require no bool and have no ability to be disabled. Such as the main scripts. Weeding those out before fetching will reduce confusion when using this function . Case: Fetching the scripts and accessing their bool pointer.

Reply:
Yes just like the above comment, you can add another member function to get the pointer then in the UI you can filter out the functions that don't have the enabled pointer, instead this way you're making an std::vector each frame your menu renders instead of passing by reference and just filtering out the scriptz that don't have the pointer.

Reply:
> Yes just like the above comment, you can add another member function to get the pointer then in the UI you can filter out the functions that don't have the enabled pointer, instead this way you're making an std::vector each frame your menu renders instead of passing by reference and just filtering out the scriptz that don't have the pointer.

I am not knowledgable of how taxing a vector creation would be to utilize each frame, but I can outsource the task to the usage thereof: 

`for (auto& s : g_script_mgr.get_scripts_data()) {
			if(s->get_enabled_pointer())
				if (ImGui::Checkbox(s->m_name.data(), s->get_enabled_pointer())) g_notification_service->push(std::string(s->m_name).append(" script"), *s->get_enabled_pointer() ? "Resumed" : "Halted");
		}
`